### PR TITLE
Gasoline

### PR DESCRIPTION
**Pull request by YakiAttaki**
## About The Pull Request
Adds gasoline, a highly flammable and volatile fuel. Can be mixed with polystyrene to create a certain sticky pyrotechnic.

## Why It's Good For The Game
Adds more chemical variety to the game, as well as providing a method with which to make a fun chemical that sticks to kids.

## Changelog
:cl:YakiAttaki
add: Added gasoline. Can be found in maints in jerry cans or created by chemistry.
/:cl:

### CODE REVIEW

**Comment by YakiAttaki** [comment]
oh my fucking god every time i do a commit the dme just completely fucks itself

File: austation/code/modules/reagents/chemistry/reagents/pyrotechnic_reagents.dm [hunk 1]
```diff
@@ -11,3 +11,24 @@
...
+	if(method == TOUCH || method == VAPOR)
+		M.adjust_fire_stacks(reac_volume / 5)
+		return
+	..()
+
+/datum/reagent/gasoline/on_mob_life(mob/living/carbon/M)
+	. = ..()
+	M.adjustToxLoss(3,0)
+	..()
+	return TRUE
```
Comment (TheFakeElon) [inline austation/code/modules/reagents/chemistry/reagents/pyrotechnic_reagents.dm:33]: Right so, `. = ..()` calls the parent proc and copies it's return value, then `..()` calls parent again AND THEN `return TRUE` completely overrides the original parent copy call.

this is broke lmao

File: austation/code/modules/reagents/chemistry/reagents/pyrotechnic_reagents.dm [hunk 2]
```diff
@@ -11,3 +11,24 @@
...
+	reagent_state = LIQUID
+	color = "#FFBF00e9"
+	taste_description = "acrid"
+
+/datum/reagent/gasoline/reaction_mob(mob/living/M, method=TOUCH, reac_volume)
+	. = ..()
+	if(method == TOUCH || method == VAPOR)
+		M.adjust_fire_stacks(reac_volume / 5)
+		return
+	..()
```
Comment (TheFakeElon) [inline austation/code/modules/reagents/chemistry/reagents/pyrotechnic_reagents.dm:27]: same here, two parent calls

**Review by TheFakeElon — CHANGES_REQUESTED** [review]
Should probably rename it to petrol or petroleum because we're not burgerland

**Comment by YakiAttaki** [comment]
> Should probably rename it to petrol or petroleum because we're not burgerland

no

File: austation/code/modules/reagents/chemistry/reagents/pyrotechnic_reagents.dm (hunk 1 above)
Comment (YakiAttaki) [reply austation/code/modules/reagents/chemistry/reagents/pyrotechnic_reagents.dm:33]: so are you going to actually suggest changes or just shittalk the code bruz

**Comment by TheFakeElon** [comment]
> no

yes

**Comment by YakiAttaki** [comment]
> > no
> 
> yes

no

**Pull request merged**